core/util/src/mill/util/Jvm.scala

```diff
@@ -219,7 +219,7 @@ object Jvm {
    */
   def createClassLoader(
       classPath: Iterable[os.Path],
-      parent: ClassLoader = null,
+      parent: ClassLoader = Thread.currentThread().getContextClassLoader,
```

Comment:
I think better to keep this default as is; flipping the default will likely cause a lot of confusion since the code will all compile and run but possibly fail with subtle classloading issues

Reply:
I'm also in favor of avoiding the context classloader if we can specify one explicitly.
Since this PR also adds `createIsolatedClassLoader`, we should just remove the default argument here.